dattri/func/ihvp.py

```diff
@@ -877,29 +893,38 @@ def ihvp_at_x_datainf(
         A function that takes a tuple `v` and returns the tuple of IHVPs of the Hessian
         of `func` and `v`.
     """
+
     def _per_sample_grad(*args) -> torch.Tensor:
         return grad(func, argnums=argnums)(*args)
 
-    def _single_datainf_ihvp(v: torch.Tensor,
-                                     grad: torch.Tensor,
-                                     regularization: float) -> torch.Tensor:
-        coef = (v.T @ grad) / (regularization + torch.sum(grad ** 2))
+    def _single_datainf_ihvp(
+        v: torch.Tensor,
+        grad: torch.Tensor,
+        regularization: float,
+    ) -> torch.Tensor:
+        coef = (v.T @ grad) / (regularization + torch.sum(grad**2))
         return (v - coef * grad) / regularization
+
     grads = vmap(_per_sample_grad, in_dims=in_dims)(*x)
     layer_cnt = len(grads)
     if param_layer_map is not None:
         grouped = []
         max_layer = max(param_layer_map)
         for group in range(max_layer + 1):
-            grouped_layers = tuple([grads[layer] for layer
-                                    in range(len(param_layer_map))
-                                    if param_layer_map[layer] == group])
+            grouped_layers = tuple(
+                [
+                    grads[layer]
+                    for layer in range(len(param_layer_map))
+                    if param_layer_map[layer] == group
+                ],
+            )
             concated_grads = torch.concat(grouped_layers, dim=1)
             grouped.append(concated_grads)
         grads = tuple(grouped)
         layer_cnt = max_layer + 1  # Assuming count starts from 0
 
-    def _ihvp_at_x_datainf_func(v: Tuple[torch.Tensor, ...],
+    def _ihvp_at_x_datainf_func(
```

Comment:
And also this function and the last part of `_ihvp_datainf_func`.
It is possible to optimize a bit?

Reply:
leave a todo for @charles-pyj